src/hw/hw_mngr.cpp

```diff
@@ -128,6 +262,74 @@ bool HWManager::setOption(const OptionType& optType, const int& instance,
     return true;
 }
 
+FanState HWManager::getDetectFanState()
+{
+    return detectFansState;
+}
+
+void HWManager::publishSystemFans()
+{
+    if (!config.desc)
+        return;
+
+    log<level::DEBUG>("HWManager::publishSystemFans()");
+
+    //  remove existing system fans before re-publish
+    for (auto it = fans.begin(); it != fans.end(); )
+    {
+        const auto& fan = *it;
+
+        size_t pos = fan->name().find(sysFanPrefixName);
+        if (pos != std::string::npos)
+        {
+            fans.erase(it);
```

Comment:
`it` may contain a non valid value after this call.
Likely `it = fans.erase(it);` should be here.

Reply:
fixed